File: clients/fireback-react/src/sdk/fireback/modules/workspaces/NotificationConfigEntity.ts
import {
    BaseDto,
    BaseEntity,
} from "../../core/definitions"
import {
    EmailProviderEntity,
} from "./EmailProviderEntity"
import {
    EmailSenderEntity,
} from "./EmailSenderEntity"
import {
    GsmProviderEntity,
} from "./GsmProviderEntity"
// In this section we have sub entities related to this object
// Class body
export type NotificationConfigEntityKeys =
  keyof typeof NotificationConfigEntity.Fields;
export class NotificationConfigEntity extends BaseEntity {
  public cascadeToSubWorkspaces?: boolean | null;
  public forcedCascadeEmailProvider?: boolean | null;
  public generalEmailProvider?: EmailProviderEntity | null;
      generalEmailProviderId?: string | null;
  public generalGsmProvider?: GsmProviderEntity | null;
      generalGsmProviderId?: string | null;
  public inviteToWorkspaceContent?: string | null;
  public inviteToWorkspaceContentExcerpt?: string | null;
  public inviteToWorkspaceContentDefault?: string | null;
  public inviteToWorkspaceContentDefaultExcerpt?: string | null;
  public inviteToWorkspaceTitle?: string | null;
  public inviteToWorkspaceTitleDefault?: string | null;
  public inviteToWorkspaceSender?: EmailSenderEntity | null;
      inviteToWorkspaceSenderId?: string | null;
  public accountCenterEmailSender?: EmailSenderEntity | null;
      accountCenterEmailSenderId?: string | null;
  public forgetPasswordContent?: string | null;
  public forgetPasswordContentExcerpt?: string | null;
  public forgetPasswordContentDefault?: string | null;
  public forgetPasswordContentDefaultExcerpt?: string | null;
  public forgetPasswordTitle?: string | null;
  public forgetPasswordTitleDefault?: string | null;
  public forgetPasswordSender?: EmailSenderEntity | null;
      forgetPasswordSenderId?: string | null;
  public acceptLanguage?: string | null;
  public confirmEmailSender?: EmailSenderEntity | null;
      confirmEmailSenderId?: string | null;
  public confirmEmailContent?: string | null;
  public confirmEmailContentExcerpt?: string | null;
  public confirmEmailContentDefault?: string | null;
  public confirmEmailContentDefaultExcerpt?: string | null;
  public confirmEmailTitle?: string | null;
  public confirmEmailTitleDefault?: string | null;
  public static Navigation = {
      edit(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/notification-config/edit/${uniqueId}`;
      },
      create(locale?: string) {
          return `${locale ? '/' + locale : ''}/notification-config/new`;
      },
      single(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/notification-config/${uniqueId}`;
      },
      query(params: any = {}, locale?: string) {
          return `${locale ? '/' + locale : ''}/notification-configs`;
      },
      /**
      * Use R series while building router in CRA or nextjs, or react navigation for react Native
      * Might be useful in Angular as well.
      **/
      Redit: "notification-config/edit/:uniqueId",
      Rcreate: "notification-config/new",
      Rsingle: "notification-config/:uniqueId",
      Rquery: "notification-configs",
  };
  public static definition = {
  "name": "notificationConfig",
  "distinctBy": "workspace",
  "http": {},
  "gormMap": {},
  "importList": [
    "modules/workspaces/GsmProviderDefinitions.dyno.proto",
    "modules/workspaces/EmailProviderDefinitions.dyno.proto",
    "modules/workspaces/EmailSenderDefinitions.dyno.proto"
  ],
  "fields": [
    {
      "name": "cascadeToSubWorkspaces",
      "type": "bool",
      "computedType": "boolean",
      "gormMap": {}
    },
    {
      "name": "forcedCascadeEmailProvider",
      "type": "bool",
      "computedType": "boolean",
      "gormMap": {}
    },
    {
      "name": "generalEmailProvider",
      "type": "one",
      "target": "EmailProviderEntity",
      "computedType": "EmailProviderEntity",
      "gormMap": {}
    },
    {
      "name": "generalGsmProvider",
      "type": "one",
      "target": "GsmProviderEntity",
      "computedType": "GsmProviderEntity",
      "gormMap": {}
    },
    {
      "name": "inviteToWorkspaceContent",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "inviteToWorkspaceContentExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "inviteToWorkspaceContentDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "inviteToWorkspaceContentDefaultExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "inviteToWorkspaceTitle",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "inviteToWorkspaceTitleDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "inviteToWorkspaceSender",
      "type": "one",
      "target": "EmailSenderEntity",
      "computedType": "EmailSenderEntity",
      "gormMap": {}
    },
    {
      "name": "accountCenterEmailSender",
      "type": "one",
      "target": "EmailSenderEntity",
      "computedType": "EmailSenderEntity",
      "gormMap": {}
    },
    {
      "name": "forgetPasswordContent",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "forgetPasswordContentExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "forgetPasswordContentDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "forgetPasswordContentDefaultExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "forgetPasswordTitle",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "forgetPasswordTitleDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "forgetPasswordSender",
      "type": "one",
      "target": "EmailSenderEntity",
      "computedType": "EmailSenderEntity",
      "gormMap": {}
    },
    {
      "name": "acceptLanguage",
      "type": "text",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "confirmEmailSender",
      "type": "one",
      "target": "EmailSenderEntity",
      "computedType": "EmailSenderEntity",
      "gormMap": {}
    },
    {
      "name": "confirmEmailContent",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "confirmEmailContentExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "confirmEmailContentDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "confirmEmailContentDefaultExcerpt",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    },
    {
      "name": "confirmEmailTitle",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "confirmEmailTitleDefault",
      "type": "string",
      "computedType": "string",
      "gormMap": {},
      "sql": "false"
    }
  ],
  "cliShort": "config",
  "cliDescription": "Configuration for the notifications used in the app, such as default gsm number, email senders, and many more"
}
public static Fields = {
  ...BaseEntity.Fields,
      cascadeToSubWorkspaces: 'cascadeToSubWorkspaces',
      forcedCascadeEmailProvider: 'forcedCascadeEmailProvider',
          generalEmailProviderId: 'generalEmailProviderId',
      generalEmailProvider$: 'generalEmailProvider',
      generalEmailProvider: EmailProviderEntity.Fields,
          generalGsmProviderId: 'generalGsmProviderId',
      generalGsmProvider$: 'generalGsmProvider',
      generalGsmProvider: GsmProviderEntity.Fields,
      inviteToWorkspaceContent: 'inviteToWorkspaceContent',
      inviteToWorkspaceContentExcerpt: 'inviteToWorkspaceContentExcerpt',
      inviteToWorkspaceContentDefault: 'inviteToWorkspaceContentDefault',
      inviteToWorkspaceContentDefaultExcerpt: 'inviteToWorkspaceContentDefaultExcerpt',
      inviteToWorkspaceTitle: 'inviteToWorkspaceTitle',
      inviteToWorkspaceTitleDefault: 'inviteToWorkspaceTitleDefault',
          inviteToWorkspaceSenderId: 'inviteToWorkspaceSenderId',
      inviteToWorkspaceSender$: 'inviteToWorkspaceSender',
      inviteToWorkspaceSender: EmailSenderEntity.Fields,
          accountCenterEmailSenderId: 'accountCenterEmailSenderId',
      accountCenterEmailSender$: 'accountCenterEmailSender',
      accountCenterEmailSender: EmailSenderEntity.Fields,
      forgetPasswordContent: 'forgetPasswordContent',
      forgetPasswordContentExcerpt: 'forgetPasswordContentExcerpt',
      forgetPasswordContentDefault: 'forgetPasswordContentDefault',
      forgetPasswordContentDefaultExcerpt: 'forgetPasswordContentDefaultExcerpt',
      forgetPasswordTitle: 'forgetPasswordTitle',
      forgetPasswordTitleDefault: 'forgetPasswordTitleDefault',
          forgetPasswordSenderId: 'forgetPasswordSenderId',
      forgetPasswordSender$: 'forgetPasswordSender',
      forgetPasswordSender: EmailSenderEntity.Fields,
      acceptLanguage: 'acceptLanguage',
          confirmEmailSenderId: 'confirmEmailSenderId',
      confirmEmailSender$: 'confirmEmailSender',
      confirmEmailSender: EmailSenderEntity.Fields,
      confirmEmailContent: 'confirmEmailContent',
      confirmEmailContentExcerpt: 'confirmEmailContentExcerpt',
      confirmEmailContentDefault: 'confirmEmailContentDefault',
      confirmEmailContentDefaultExcerpt: 'confirmEmailContentDefaultExcerpt',
      confirmEmailTitle: 'confirmEmailTitle',
      confirmEmailTitleDefault: 'confirmEmailTitleDefault',
}
}

File: modules/fireback/codegen/ios/firebackios/modules/workspaces/NotificationConfigEntity.swift
class NotificationConfigEntity : Codable {
    private var _cascadeToSubWorkspaces: Bool?
    var `CascadeToSubWorkspaces`: Bool? {
        set { _cascadeToSubWorkspaces = newValue }
        get { return _cascadeToSubWorkspaces }
    }
    private var _forcedCascadeEmailProvider: Bool?
    var `ForcedCascadeEmailProvider`: Bool? {
        set { _forcedCascadeEmailProvider = newValue }
        get { return _forcedCascadeEmailProvider }
    }
    private var _generalEmailProvider: EmailProviderEntity?
    var `GeneralEmailProvider`: EmailProviderEntity? {
        set { _generalEmailProvider = newValue }
        get { return _generalEmailProvider }
    }
    private var _generalGsmProvider: GsmProviderEntity?
    var `GeneralGsmProvider`: GsmProviderEntity? {
        set { _generalGsmProvider = newValue }
        get { return _generalGsmProvider }
    }
    private var _inviteToWorkspaceContent: String?
    var `InviteToWorkspaceContent`: String? {
        set { _inviteToWorkspaceContent = newValue }
        get { return _inviteToWorkspaceContent }
    }
    private var _inviteToWorkspaceContentExcerpt: String?
    var `InviteToWorkspaceContentExcerpt`: String? {
        set { _inviteToWorkspaceContentExcerpt = newValue }
        get { return _inviteToWorkspaceContentExcerpt }
    }
    private var _inviteToWorkspaceContentDefault: String?
    var `InviteToWorkspaceContentDefault`: String? {
        set { _inviteToWorkspaceContentDefault = newValue }
        get { return _inviteToWorkspaceContentDefault }
    }
    private var _inviteToWorkspaceContentDefaultExcerpt: String?
    var `InviteToWorkspaceContentDefaultExcerpt`: String? {
        set { _inviteToWorkspaceContentDefaultExcerpt = newValue }
        get { return _inviteToWorkspaceContentDefaultExcerpt }
    }
    private var _inviteToWorkspaceTitle: String?
    var `InviteToWorkspaceTitle`: String? {
        set { _inviteToWorkspaceTitle = newValue }
        get { return _inviteToWorkspaceTitle }
    }
    private var _inviteToWorkspaceTitleDefault: String?
    var `InviteToWorkspaceTitleDefault`: String? {
        set { _inviteToWorkspaceTitleDefault = newValue }
        get { return _inviteToWorkspaceTitleDefault }
    }
    private var _inviteToWorkspaceSender: EmailSenderEntity?
    var `InviteToWorkspaceSender`: EmailSenderEntity? {
        set { _inviteToWorkspaceSender = newValue }
        get { return _inviteToWorkspaceSender }
    }
    private var _forgetPasswordContent: String?
    var `ForgetPasswordContent`: String? {
        set { _forgetPasswordContent = newValue }
        get { return _forgetPasswordContent }
    }
    private var _forgetPasswordContentExcerpt: String?
    var `ForgetPasswordContentExcerpt`: String? {
        set { _forgetPasswordContentExcerpt = newValue }
        get { return _forgetPasswordContentExcerpt }
    }
    private var _forgetPasswordContentDefault: String?
    var `ForgetPasswordContentDefault`: String? {
        set { _forgetPasswordContentDefault = newValue }
        get { return _forgetPasswordContentDefault }
    }
    private var _forgetPasswordContentDefaultExcerpt: String?
    var `ForgetPasswordContentDefaultExcerpt`: String? {
        set { _forgetPasswordContentDefaultExcerpt = newValue }
        get { return _forgetPasswordContentDefaultExcerpt }
    }
    private var _forgetPasswordTitle: String?
    var `ForgetPasswordTitle`: String? {
        set { _forgetPasswordTitle = newValue }
        get { return _forgetPasswordTitle }
    }
    private var _forgetPasswordTitleDefault: String?
    var `ForgetPasswordTitleDefault`: String? {
        set { _forgetPasswordTitleDefault = newValue }
        get { return _forgetPasswordTitleDefault }
    }
    private var _forgetPasswordSender: EmailSenderEntity?
    var `ForgetPasswordSender`: EmailSenderEntity? {
        set { _forgetPasswordSender = newValue }
        get { return _forgetPasswordSender }
    }
    private var _acceptLanguage: String?
    var `AcceptLanguage`: String? {
        set { _acceptLanguage = newValue }
        get { return _acceptLanguage }
    }
    private var _confirmEmailSender: EmailSenderEntity?
    var `ConfirmEmailSender`: EmailSenderEntity? {
        set { _confirmEmailSender = newValue }
        get { return _confirmEmailSender }
    }
    private var _confirmEmailContent: String?
    var `ConfirmEmailContent`: String? {
        set { _confirmEmailContent = newValue }
        get { return _confirmEmailContent }
    }
    private var _confirmEmailContentExcerpt: String?
    var `ConfirmEmailContentExcerpt`: String? {
        set { _confirmEmailContentExcerpt = newValue }
        get { return _confirmEmailContentExcerpt }
    }
    private var _confirmEmailContentDefault: String?
    var `ConfirmEmailContentDefault`: String? {
        set { _confirmEmailContentDefault = newValue }
        get { return _confirmEmailContentDefault }
    }
    private var _confirmEmailContentDefaultExcerpt: String?
    var `ConfirmEmailContentDefaultExcerpt`: String? {
        set { _confirmEmailContentDefaultExcerpt = newValue }
        get { return _confirmEmailContentDefaultExcerpt }
    }
    private var _confirmEmailTitle: String?
    var `ConfirmEmailTitle`: String? {
        set { _confirmEmailTitle = newValue }
        get { return _confirmEmailTitle }
    }
    private var _confirmEmailTitleDefault: String?
    var `ConfirmEmailTitleDefault`: String? {
        set { _confirmEmailTitleDefault = newValue }
        get { return _confirmEmailTitleDefault }
    }
}